Realize inputs to DynamicScalar before unwrapping storage

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #118411
* __->__ #118125
* #117862

Fixes https://github.com/pytorch/pytorch/issues/118102

Unfortunately, the test still fails due to an unrelated problem https://github.com/pytorch/pytorch/issues/117665

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

ezyang committed:
Realize inputs to DynamicScalar before unwrapping storage

Unfortunately, the test still fails due to an unrelated problem
https://github.com/pytorch/pytorch/issues/117665

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

[ghstack-poisoned]

ezyang commented on torch/_inductor/lowering.py:
```diff
@@ -2611,6 +2611,8 @@ def inner(
         if len(size) == 1 and isinstance(size[0], (list, tuple, torch.Size)):
             size = tuple(size[0])
+        for s in size:
+            assert not isinstance(s, torch.SymInt)
```
This makes the later problem trigger more clearly @eellison

eellison replied:
Maybe add a comment or update assertion along the lines of `should be Symbol not SymInt`

jansel reviewed (dismissed):
Test failures?

ezyang committed:
Update on "Realize inputs to DynamicScalar before unwrapping storage"


Fixes https://github.com/pytorch/pytorch/issues/118102

Unfortunately, the test still fails due to an unrelated problem https://github.com/pytorch/pytorch/issues/117665

Signed-off-by: Edward Z. Yang <ezyangmeta.com>

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

ezyang commented:
@jansel just a dumb typo for the expectedFailure lol

ezyang committed:
Update on "Realize inputs to DynamicScalar before unwrapping storage"


Fixes https://github.com/pytorch/pytorch/issues/118102

Unfortunately, the test still fails due to an unrelated problem https://github.com/pytorch/pytorch/issues/117665

Signed-off-by: Edward Z. Yang <ezyangmeta.com>

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge

ezyang committed:
Update on "Realize inputs to DynamicScalar before unwrapping storage"


Fixes https://github.com/pytorch/pytorch/issues/118102

Unfortunately, the test still fails due to an unrelated problem https://github.com/pytorch/pytorch/issues/117665

Signed-off-by: Edward Z. Yang <ezyangmeta.com>

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge

ezyang commented:
@pytorchbot merge -i

ezyang commented:
@pytorchbot merge -i
